src/main/java/org/apache/maven/plugins/dependency/tree/VerboseGraphSerializer.java

```diff
@@ -42,30 +42,370 @@
             MANAGED_SCOPE = "managedScope";
 
     public String serialize( DependencyNode root )
+    {
+        return serialize( root, "" );
+    }
+
+    public String serialize( DependencyNode root, String outputType )
     {
         Set<String> coordinateStrings = new HashSet<>();
         Map<String, String> coordinateVersionMap = new HashMap<>();
-        StringBuilder builder = new StringBuilder();
 
         // Use BFS to mirror how Maven resolves dependencies and use DFS to print the tree easily
         Map<DependencyNode, String> nodeErrors = getNodeConflictMessagesBfs( root, coordinateStrings,
                 coordinateVersionMap );
 
+        if ( "graphml".equals( outputType ) )
+        {
+            return serializeGraphml( root, nodeErrors );
+        }
+        else if ( "tgf".equals( outputType ) )
+        {
+            return serializeTgf( root, nodeErrors );
+        }
+        else if ( "dot".equals( outputType ) )
+        {
+            return serializeDot( root, nodeErrors );
+        }
+        else
+        {
+            return serializeText( root, nodeErrors );
+        }
+    }
+
+    private String serializeGraphml( DependencyNode root, Map<DependencyNode, String> nodeErrors )
+    {
+        Set<DependencyNode> visitedNodes = new HashSet<>();
+        Queue<DependencyNode> queue = new LinkedList<>();
+        queue.add( root );
+        StringBuilder result = new StringBuilder( "<?xml version=\"1.0\" encoding=\"UTF-8\"?> "
+                        + "<graphml xmlns=\"http://graphml.graphdrawing.org/xmlns\" "
+                + "xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" "
+                + "xmlns:y=\"http://www.yworks.com/xml/graphml\" "
+                + "xsi:schemaLocation=\"http://graphml.graphdrawing.org/xmlns "
+                + "http://graphml.graphdrawing.org/xmlns/1.0/graphml.xsd\">"
+                + System.lineSeparator()
+                + "  <key for=\"node\" id=\"d0\" yfiles.type=\"nodegraphics\"/>"
+                + System.lineSeparator()
+                + "  <key for=\"edge\" id=\"d1\" yfiles.type=\"edgegraphics\"/>"
+                + System.lineSeparator()
```

Comment:
to make this reproducible use a standard line separator, probably \n. No reason to use System line separator when writing XML in 2020.